src/sentry/models/organization.py

```diff
@@ -122,49 +122,31 @@ def get_for_user(self, user, scope=None, only_visible=True):
             return [r.organization for r in results if scope in r.get_scopes()]
         return [r.organization for r in results]
 
-    def get_organizations_where_user_is_owner(
-        self, user_id: int, queryset: QuerySet = None
-    ) -> QuerySet:
+    def get_organizations_where_user_is_owner(self, user_id: int) -> QuerySet:
         """
         Returns a QuerySet of all organizations where a user has the top priority role.
         The default top priority role in Sentry is owner.
         """
 
-        # get orgs and orgmemberIDs for the user
-        query = OrganizationMember.objects.filter(user_id=user_id)
-        if queryset:  # queryset is a QuerySet of valid orgs
-            orgs_and_members = query.filter(organization__in=queryset)
-        orgs_and_members = query.values_list("organization_id", "id")
+        orgs = Organization.objects.filter(
+            member_set__user_id=user_id,
+            status=OrganizationStatus.VISIBLE,
+        )
 
-        organizations, org_members = zip(*orgs_and_members)
-        organizations = list(organizations)
-        org_members = list(org_members)
+        # get owners from orgs
+        owner_role_orgs = orgs.filter(member_set__role=roles.get_top_dog().id)
 
         # get owner teams
         owner_teams = Team.objects.filter(
-            organization_id__in=organizations, org_role=roles.get_top_dog().id
+            organization__in=orgs, org_role=roles.get_top_dog().id
         ).values_list("id", flat=True)
 
-        # get owners from owner teams
-        owners = list(
-            OrganizationMemberTeam.objects.filter(
-                team_id__in=owner_teams,
-                organizationmember_id__in=org_members,
-            ).values_list("organizationmember_id", flat=True)
-        )
-
-        # get corresponding orgs from org list
-        members_to_org = {m: organizations[i] for (i, m) in enumerate(org_members)}
-        orgs = {members_to_org[m] for m in owners if m in members_to_org}
+        # get the orgs in which the user is a member of an owner team
+        owner_team_member_orgs = OrganizationMemberTeam.objects.filter(
+            team_id__in=owner_teams
+        ).values_list("organizationmember__organization_id", flat=True)
 
-        # get owners from orgs
-        owner_role_orgs = set(
-            OrganizationMember.objects.filter(
-                role=roles.get_top_dog().id, user_id=user_id
-            ).values_list("organization_id", flat=True)
-        )
-
-        return self.filter(id__in=orgs.union(owner_role_orgs), status=OrganizationStatus.ACTIVE)
+        return self.filter(id__in=owner_team_member_orgs).union(owner_role_orgs)
```

Comment:
Same here if you want to evaluate before making the query, wrap in `list`. I'm not sure that you need to though, I think the union will work fine

Reply:
Yeah, I think it's okay without wrapping in `list`. The problem before wasn't that we were nesting queries as such, but that one of the nested queries was returning the world. As long as every query set has some sane filters, the structure here should be fine.